lib/api/apiUtils/object/storeObject.js

```diff
@@ -26,7 +26,9 @@ function checkHashMatchMD5(stream, hashedStream, dataRetrievalInfo, log, cb) {
             contentMD5,
         });
         log.trace('contentMD5 does not match, deleting data');
```

Comment:
In this case, it turns out to be a bigger issue than I though since a user could create orphan data on the server just by sending malformed requests. But again I'm not sure what is this actual code checking.